packages/binding/src/definition/definition.ts

```diff
@@ -144,8 +194,17 @@ const getPossibleElement = (param: {
       if (!constParam.type) {
         continue;
       }
-      const reference = getReference(constParam.type);
       const name = sorterMap.get(constParam.name) ?? constParam.name;
+      if (name === "vSorterInfo" && constParam.parameterProperties) {
```

Comment:
Having custom logic based on a name doesn't seem right. Why do we have problems handling `string|object` type? I suspect we would have similar issues elsewhere, because I think it is somewhat common pattern used in UI5.

Reply:
The issue here is how sorter is initialized. In Javascript, user can pass `vSorterInfo` as param to constructor of Sorter. e.g 
```javascript
new sap.ui.model.Sorter(vSorterInfo)
```
where `vSorterInfo` can have properties e.g `comparator`. This syntax is fine in Javascript, but not ok in XML syntax. This `vSorterInfo` information is added to  [design time api.json ](https://ui5.sap.com/1.131.0/test-resources/sap/ui/core/designtime/api.json) which is the root cause of this issue. 

It would be ideal to have clean data to work with, but this is not always the case.

Reply:
ok, I see. There is a specific logic how the constructor is called when the XML template is processed and it does not necessarily match the generic way how it can be called from JS.